src/shared/particle_dynamics/particle_functors.h

```diff
@@ -199,13 +204,41 @@ class LinearGradientCorrection : public KernelCorrection
         : KernelCorrection(),
           B_(*particles->getVariableDataByName<Matd>("LinearGradientCorrectionMatrix")){};
 
+    Matd operator()(size_t index_i, size_t index_j)
+    {
+        return B_[index_i];
+    };
+
+    Matd operator()(size_t index_i)
+    {
+        return B_[index_i];
+    };
+
+  protected:
+    StdLargeVec<Matd> &B_;
+};
+
+class LinearGradientCorrectionWithBulkScope : public KernelCorrection
+{
+  public:
+    LinearGradientCorrectionWithBulkScope(BaseParticles *particles)
+        : KernelCorrection(),
+          B_(*particles->getVariableDataByName<Matd>("LinearGradientCorrectionMatrix")),
+          within_scope_(particles){};
+
+    Matd operator()(size_t index_j, size_t index_i)
+    {
```

Comment:
I got it. This is the answer to my concern. There is a selection between i and j.

Reply:
This is not satisfactory. However, at present no other good way to incorporate the change.
Actually, this case raise a issue on how to handle boundary like region within a body.  it is not other body, not ghost and need to handle separately,